scripts/common_startup.sh

```diff
@@ -224,14 +224,23 @@ if [ $SKIP_CLIENT_BUILD -eq 0 ]; then
     fi
 fi
 
+# Install node if not installed
+if [ -n "$VIRTUAL_ENV" ]; then
+    if ! in_venv "$(command -v node)" || [ "$(node --version)" != "v$NODE_VERSION" ]; then
+        echo "Installing node into $VIRTUAL_ENV with nodeenv."
+        nodeenv -n $NODE_VERSION -p
+    fi
+elif [ -n "$CONDA_DEFAULT_ENV" -a -n "$CONDA_EXE" ]; then
```

Comment:
This change seems unrelated to the PR and I am a bit reluctant, since the things we do with startup and node has shown to be troublesome for various different-than-ours environments. Maybe we could extract it to a separate PR for more eyes to see? I might be overstating the risk though.

Reply:
Is this in regards to the conda piece specifically or moving the whole block including the nodeenv thing outside the build client block below?

Reply:
The conda part mostly.

Trying to install node every time with no way for the admin/user to opt out (unless you stop using common_startup/virtualenv) could also be tricky based on the number of reports/feedback we got with client builds lately but that is our new requirement after this PR and I am fine with that and it is early enough in the dev cycle to introduce it.

I understand we now need it even when client build is skipped

Reply:
It is @mvdbeek's commit but I like it as it is and 👍 it. I feel like this would be the correct way to handle installing node in a Conda environment and that is pretty experimental so I'm not too worried about breaking it - seems more likely to fix something than break it. Unrelated sure, but who would comment on Conda dependency installation - probably only @mvdbeek anyway 😅.

As for this not being hidden behind a flag, I suspect at some point someone will ask for a flag (given https://github.com/galaxyproject/galaxy/commit/6e865a3363587582165788ea7b604cda084dff4f it will probably be me) but I'd wait until someone asks to add that logic because we could be wrong and hopefully are.

Reply:
I see, I did not notice that `CONDA_EXE` is only when we use conda to install deps.  resolved!

Reply:
Yeah, just wanted to make sure that by default we'd get node installed, which was causing the circle-ci builds to fail. If you guys want to take this one out and instead raise a skip in the unit test if there's no node that's fine with me as well.